### ISSUE TITLE
False positive on CO_ABSTRACT_SELF ?

### ISSUE THREAD

**Issue by lqc** [spotbugs/spotbugs#454]
It seems that this bug is reported even if the class implementing the `compareTo()` does not inherit from `Comparable`, see https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L71 for an example.

I'm not sure if this is a false positive or author's intention. If it is by design, then the message at least is confusing:

```
This class defines a covariant version of compareTo().  To correctly override the compareTo() method in the Comparable interface, the parameter of compareTo() must have type java.lang.Object.
```

as the example class doesn't have anything to do with `Comparable` interface.

**Comment by jolkdarr** [comment]
Hello.
In your example, is the issue reported on the class `ComparableVersion`?

**Comment by lqc** [comment]
Yes, specifically on the "Item" inner class:

```
Abstract package.name.ComparableVersion$Item defines compareTo(ComparableVersion$Item) method
```

**Comment by jolkdarr** [comment]
If the rule checks the first `compareTo` method "within" the outer class, it's actually a bug.
Maybe there's no warning if `compareTo` is declared firstly.

**Comment by lqc** [comment]
Not yet. Now that I see where that there indeed is a violation I'll try to fix this.
